src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala

```diff
@@ -290,6 +290,65 @@ class PodsControllerTest extends UnitTest with ScalatestRouteTest with RouteBeha
     //        response.getMetadata.containsKey(RestResource.DeploymentHeader) should be(true)
     //      }
     //    }
+
+    "update a simple single-container pod from docker image w/ shell command" in {
+      implicit val podSystem = mock[PodManager]
+      val f = Fixture()
+      val controller = f.controller()
+
+      val deploymentPlan = DeploymentPlan.empty
+      f.podManager.update(any, eq(false)).returns(Future.successful(deploymentPlan))
+
+      val postJson = """
+                       | { "id": "/mypod", "networks": [ { "mode": "host" } ], "containers": [
+                       |   { "name": "webapp",
+                       |     "resources": { "cpus": 0.03, "mem": 64 },
+                       |     "image": { "kind": "DOCKER", "id": "busybox" },
+                       |     "exec": { "command": { "shell": "sleep 1" } } } ] }
+                     """.stripMargin
+      val entity = HttpEntity(postJson).withContentType(ContentTypes.`application/json`)
+      val request = Put("/mypod")
+        .withEntity(entity)
+        .withHeaders(`Remote-Address`(RemoteAddress(InetAddress.getByName("192.168.3.12"))))
+
+      request ~> controller.route ~> check {
+        response.status should be(StatusCodes.OK)
+        response.header[Headers.`Marathon-Deployment-Id`].value.value() should be(deploymentPlan.id)
+      }
+    }
+
+    "save pod with more than one instance" in {
+      implicit val podSystem = mock[PodManager]
+      val f = Fixture()
+      val controller = f.controller()
+
+      val deploymentPlan = DeploymentPlan.empty
+      f.podManager.update(any, eq(false)).returns(Future.successful(deploymentPlan))
+
+      val postJson = """
+                           | { "id": "/mypod", "networks": [ { "mode": "host" } ],
+                           | "scaling": { "kind": "fixed", "instances": 2 }, "containers": [
+                           |   { "name": "webapp",
+                           |     "resources": { "cpus": 0.03, "mem": 64 },
+                           |     "exec": { "command": { "shell": "sleep 1" } } } ] }
+                         """.stripMargin
+
+      val entity = HttpEntity(postJson).withContentType(ContentTypes.`application/json`)
+      val request = Put("/mypod")
+        .withEntity(entity)
+        .withHeaders(`Remote-Address`(RemoteAddress(InetAddress.getByName("192.168.3.12"))))
+
+      request ~> controller.route ~> check {
+        response.status should be(StatusCodes.OK)
+        response.header[Headers.`Marathon-Deployment-Id`].value.value() should be(deploymentPlan.id)
+
+        val pod = Json.parse(responseAs[String]).as[raml.Pod]
```

Comment:
Check out the tests in https://github.com/mesosphere/marathon/pull/5741. I've added have matchers for json responses that make testing a little cleaner.